packages/expo-updates/ios/EXUpdates/EXUpdatesDevLauncherController.m

```diff
@@ -0,0 +1,119 @@
+
+@synthesize bridge = _bridge;
+
++ (instancetype)sharedInstance
+{
+  static EXUpdatesDevLauncherController *theController;
+  static dispatch_once_t once;
+  dispatch_once(&once, ^{
+    if (!theController) {
+      theController = [EXUpdatesDevLauncherController new];
+    }
+  });
+  return theController;
+}
+
+- (void)setBridge:(nullable id)bridge
+{
+  _bridge = bridge;
+  if ([bridge isKindOfClass:[RCTBridge class]]) {
+    EXUpdatesAppController.sharedInstance.bridge = (RCTBridge *)bridge;
+  }
+}
+
+- (NSURL *)launchAssetURL
+{
+  return EXUpdatesAppController.sharedInstance.launchAssetUrl;
+}
+
+- (void)fetchUpdateWithConfiguration:(NSDictionary *)configuration
+                          onManifest:(EXUpdatesManifestBlock)manifestBlock
+                            progress:(EXUpdatesProgressBlock)progressBlock
+                             success:(EXUpdatesSuccessBlock)successBlock
+                               error:(EXUpdatesErrorBlock)errorBlock
+{
+  EXUpdatesAppController *controller = EXUpdatesAppController.sharedInstance;
+  EXUpdatesConfig *updatesConfiguration = [EXUpdatesConfig configWithExpoPlist];
+  [updatesConfiguration loadConfigFromDictionary:configuration];
+  if (!updatesConfiguration.updateUrl) {
+    errorBlock([NSError errorWithDomain:EXUpdatesDevLauncherControllerErrorDomain code:1 userInfo:@{NSLocalizedDescriptionKey: @"Failed to load update: configuration object must include a valid update URL"}]);
+    return;
+  }
+  NSError *fsError;
+  if (![controller initializeUpdatesDirectoryWithError:&fsError]) {
+    errorBlock(fsError ?: [NSError errorWithDomain:EXUpdatesDevLauncherControllerErrorDomain code:2 userInfo:@{NSLocalizedDescriptionKey: @"Failed to initialize updates directory with an unknown error"}]);
+    return;
+  }
+  NSError *dbError;
+  if (![controller initializeUpdatesDatabaseWithError:&dbError]) {
+    errorBlock(dbError ?: [NSError errorWithDomain:EXUpdatesDevLauncherControllerErrorDomain code:3 userInfo:@{NSLocalizedDescriptionKey: @"Failed to initialize updates database with an unknown error"}]);
+    return;
+  }
+
+  [controller setIsStarted:YES];
+  [self _setDevelopmentSelectionPolicy];
+
+  EXUpdatesRemoteAppLoader *loader = [[EXUpdatesRemoteAppLoader alloc] initWithConfig:updatesConfiguration database:controller.database directory:controller.updatesDirectory completionQueue:controller.controllerQueue];
+  [loader loadUpdateFromUrl:updatesConfiguration.updateUrl onManifest:^BOOL(EXUpdatesUpdate * _Nonnull update) {
+    return manifestBlock(update.rawManifest.rawManifestJSON);
+  } asset:^(EXUpdatesAsset * _Nonnull asset, NSUInteger successfulAssetCount, NSUInteger failedAssetCount, NSUInteger totalAssetCount) {
+    progressBlock(successfulAssetCount, failedAssetCount, totalAssetCount);
+  } success:^(EXUpdatesUpdate * _Nullable update) {
+    if (update) {
+      [self _launchUpdate:update withConfiguration:updatesConfiguration success:successBlock error:errorBlock];
+    }
```

Comment:
If there is no update, we probably want to call one of the blocks (success or error) so that the code does not hang forever.

Reply:
Making this change in a separate PR since it touches both platforms: https://github.com/expo/expo/pull/13136